src/targets/maven.ts

```diff
@@ -0,0 +1,250 @@
+import { TargetConfig } from '../schemas/project_config';
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { BaseTarget } from './base';
+import { ConfigurationError } from '../utils/errors';
+import { homedir } from 'os';
+import * as path from 'path';
+import * as fs from 'fs';
```

Comment:
Seems like you only use `promises` and `readdirSync`
```suggestion
import {promises, readdirSync} from 'fs';
```